CustomCraftSML/Serialization/AliasRecipe.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace CustomCraft2SML.Serialization
+{
+    using Common;
+    using System;
+    using System.Collections.Generic;
+    using Common.EasyMarkup;
+    using CustomCraft2SML.Interfaces;
+
+    internal class AliasRecipe : AddedRecipe, IAliasRecipe
+    {
+        private readonly EmProperty<string> displayName;
+        private readonly EmProperty<string> tooltip;
+
+        public string ItemName
+        {
+            get => emTechType.Value;
+        }
+
+        public string DisplayName
+        {
+            get => displayName.Value;
+            set => displayName.Value = value;
+        }
+
+        public string Tooltip
+        {
+            get => tooltip.Value;
+            set => tooltip.Value = value;
+        }
+
+        public override TechType ItemID
+        {
+            get
+            {
+                if (internalId == TechType.None)
+                {
+                    string tAlias = String.Copy(emTechType.Value);
+                    if (tAlias.Length > 1)
+                    {
+                        internalId = SMLHelper.V2.Handlers.TechTypeHandler.AddTechType(tAlias, DisplayName, Tooltip);
```

Comment:
Would it be possible to delay interactions with SMLHelper until the `CustomCraft` class?  
I went through some effort to preserve a layer of separation between the general CC2 classes and their ultimate interactions with SMLHelper.

It might not be possible, but design-wise, the classes in the `CustomCraft2SML.Serialization` namespace were intended to only deal with serialization and deserialization.

I see you already enhanced [EmPropertyTechType](https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/blob/master/CustomCraftSML/Serialization/EmPropertyTechType.cs) to be able to handle both custom and original TechTypes.  
If there is no other way, perhaps this SMLHelper call can be handled in `EmPropertyTechType` instead of here.

Reply:
Part of the problem I had with delaying until the CustomCraft object was the initial load in the FileReaderWriter.ParseEntries.  This module does duplicate checks and needed full TechType resolution prior to that point.  It may be possible to expose the emTechType as a string and make duplicate checks on that value instead of the TechType.  Preferences?

Reply:
I'll need to give this some more thought then.  
I knew this would come up at some point because the original design was not accounting for modded TechTypes. But this feature basically requires that this issue be solved, and be solved for both the key item and also ingredient items.

It may be necessary to front load some of the `TechType` handling specifically.  
At least the recipe patching itself can be delayed.

Reply:
I was thinking about this a bit and came to one possibility.  The serializers could not have any SN/SML knowledge at all - just lists of basic types.  The duplicate check could really be done in the CustomCraft module just as easily as the ReaderWriter, though the code would just skip the duplicate rather than not add to the list.

Delaying can also solve the issue where you create a new TechType (via alias) and then use it in a recipe later in the file.  Currently, the ingredient serializers will attempt to resolve the TechType name immediately on serialization, but we don't create the TechTypes for the new recipe until after all recipes are parsed and checking for duplicates.

I'm busy much of the day, but may give this a shot later.